processor/src/main/java/com/infinum/jsonapix/processor/specs/RelationshipsSpecBuilder.kt

```diff
@@ -90,6 +102,27 @@ internal object RelationshipsSpecBuilder {
             .build()
     }
 
+    private fun PropertySpec.Builder.addSerialNameAnnotationIfItExits(originalProperty: PropertySpec): PropertySpec.Builder = this.apply {
+        originalProperty.annotations.findAnnotationWithTypeName(serialNameTypeName)?.let { annotationSpec ->
+            addAnnotation(annotationSpec)
+        }
+    }
+
+    private fun mapPropertiesToParams(properties: List<PropertySpec>): List<ParameterSpec> {
+        return properties.map {
+            ParameterSpec.builder(it.name, it.type)
+                .apply {
+                    if (it.annotations.missingTypeName(serialNameTypeName)) {
```

Comment:
If the property has a @SerialName annotation, it is added automatically and we do not need to add it manually.